tags-config-service-impl/src/main/java/org/hypertrace/tag/config/service/ConfigServiceCoordinatorImpl.java

```diff
@@ -0,0 +1,113 @@
+import org.hypertrace.config.service.v1.GetConfigResponse;
+import org.hypertrace.config.service.v1.UpsertConfigRequest;
+import org.hypertrace.config.service.v1.UpsertConfigResponse;
+import org.hypertrace.core.grpcutils.client.RequestContextClientCallCredsProviderFactory;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.tag.config.service.v1.Tag;
+
+public class ConfigServiceCoordinatorImpl implements ConfigServiceCoordinator {
+  private final ConfigServiceBlockingStub configServiceBlockingStub;
+
+  public ConfigServiceCoordinatorImpl(Channel configChannel) {
+    this.configServiceBlockingStub =
+        ConfigServiceGrpc.newBlockingStub(configChannel)
+            .withCallCredentials(
+                RequestContextClientCallCredsProviderFactory.getClientCallCredsProvider().get());
+  }
+
+  @Override
+  public Tag upsertTag(RequestContext requestContext, Tag tag) {
+    UpsertConfigRequest upsertConfigRequest =
+        UpsertConfigRequest.newBuilder()
+            .setResourceName(TAG_CONFIG)
+            .setResourceNamespace(TAG_RESOURCE_NAMESPACE)
+            .setConfig(convertToGenericFromTag(tag))
+            .setContext(tag.getId())
+            .build();
+    UpsertConfigResponse upsertConfigResponse = upsertConfig(requestContext, upsertConfigRequest);
+    return convertToTagFromGeneric(upsertConfigResponse.getConfig());
+  }
+
+  public UpsertConfigResponse upsertConfig(RequestContext context, UpsertConfigRequest request) {
+    return context.call(() -> configServiceBlockingStub.upsertConfig(request));
+  }
+
+  @Override
+  public Tag getTag(RequestContext requestContext, String tagId) {
+    GetConfigRequest getConfigRequest =
+        GetConfigRequest.newBuilder()
+            .setResourceName(TAG_CONFIG)
+            .setResourceNamespace(TAG_RESOURCE_NAMESPACE)
+            .addContexts(tagId)
+            .build();
+    GetConfigResponse getConfigResponse = getConfig(requestContext, getConfigRequest);
+    return convertToTagFromGeneric(getConfigResponse.getConfig());
+  }
+
+  public GetConfigResponse getConfig(RequestContext context, GetConfigRequest request) {
+    return context.call(() -> configServiceBlockingStub.getConfig(request));
+  }
+
+  @Override
+  public List<Tag> getAllTags(RequestContext requestContext) {
+    GetAllConfigsRequest getAllConfigsRequest =
+        GetAllConfigsRequest.newBuilder()
+            .setResourceName(TAG_CONFIG)
+            .setResourceNamespace(TAG_RESOURCE_NAMESPACE)
+            .build();
+    GetAllConfigsResponse getAllConfigsResponse =
+        getAllConfigs(requestContext, getAllConfigsRequest);
+    return getAllConfigsResponse.getContextSpecificConfigsList().stream()
+        .map(configResponse -> convertToTagFromGeneric(configResponse.getConfig()))
+        .collect(Collectors.toList());
+  }
+
+  public GetAllConfigsResponse getAllConfigs(RequestContext context, GetAllConfigsRequest request) {
```

Comment:
Does this method need to be public? I don't see anyone using it.

Reply:
Yes will make it private.